File: src/Models/Buildable/House.java
package Models.Buildable;

import Models.Measurement;

import java.util.Vector;

public class House implements Buildable {
    private Wall wall;

    public House addWall(Measurement wallLength, Measurement wallHeight) throws IllegalArgumentException {
        this.wall = new Wall(wallLength, wallHeight);
        return this;
    }

    @Override
    public Vector<Vector<String>> materials() throws NullPointerException {
        return wall.materials();
    }
}

File: src/main/kotlin/models/buildable/House.kt
package models.buildable

import graphics.GraphicsList
import models.Measurement
import models.buildable.material.MaterialList
import java.util.*

class House : Buildable {
    private var wall: Wall? = null

    @Throws(IllegalArgumentException::class)
    fun addWall(wallLength: Measurement, wallHeight: Measurement): House {
        wall = Wall(wallLength, wallHeight)
        return this
    }

    @Throws(NullPointerException::class)
    override fun materials(): Vector<Vector<String>> {
        return wall?.materials() ?: MaterialList().materials()
    }
}